### ISSUE TITLE
Failed to use await all in `native-mt`

### ISSUE THREAD

**Issue by e5l** [Kotlin/kotlinx.coroutines#2025]
Using of awaitAll in frozen context leads to the exception in ktor:
```
kotlin.native.concurrent.InvalidMutabilityException: mutation attempt of frozen kotlinx.coroutines.AwaitAll.AwaitAllNode@3579ac8
	kfun:kotlin.Throwable#<init>(kotlin.String?){} (0x3913b7)
	kfun:kotlin.Exception#<init>(kotlin.String?){} (0x38ad15)
	kfun:kotlin.RuntimeException#<init>(kotlin.String?){} (0x38a8d5)
	kfun:kotlin.native.concurrent.InvalidMutabilityException#<init>(kotlin.String){} (0x3b8a85)
	ThrowInvalidMutabilityException (0x3ba183)
	MutationCheck (0xabb6ee)
	kfun:kotlinx.coroutines.AwaitAll.AwaitAllNode.<set-handle>#internal (0x480b30)
	kfun:kotlinx.coroutines.AwaitAll.await#internal (0x47fb0f)
	kfun:kotlinx.coroutines.$awaitAllCOROUTINE$534#invokeSuspend(kotlin.Result<kotlin.Any?>){}kotlin.Any? (0x47ef1b)
	kfun:kotlinx.coroutines#awaitAll@kotlin.collections.Collection<kotlinx.coroutines.Deferred<0:0>>(){0§<kotlin.Any?>}kotlin.collections.List<0:0> (0x47f1ba)
	kfun:io.ktor.util.$split$lambda-1COROUTINE$478.invokeSuspend#internal (0x5fffaf)
	kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} (0x3ad338)
	kfun:kotlinx.coroutines.DispatchedTask#run(){} (0x4e10ba)
	kfun:kotlinx.coroutines.EventLoopImplBase#processNextEvent(){}kotlin.Long (0x49477e)
	kfun:kotlinx.coroutines#runEventLoop(kotlinx.coroutines.EventLoop?;kotlin.Function0<kotlin.Boolean>){} (0x508abd)
	kfun:kotlinx.coroutines.BlockingCoroutine.joinBlocking#internal (0x509bdb)
	kfun:kotlinx.coroutines#runBlocking(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,0:0>){0§<kotlin.Any?>}0:0 (0x5091db)
```

**Comment by BoD** [comment]
See also https://github.com/ktorio/ktor/issues/1550

**Comment by qwwdfsad** [comment]
Fixed in `1.3.9-native-mt-2`
